src/octotree.js

```diff
@@ -24,9 +25,26 @@ $(document).ready(() => {
       if (event.target === window) layoutChanged();
     });
 
-    $toggler.click(toggleSidebarAndSave);
+    $document.click((event) => !isSidebarPinned() && isOutsideSidebar(event.target) && toggleSidebar(false));
+
+    $pinner.click(togglePin);
+
+    $sidebar.on(
+      `keyup mouseleave mousemove ${EVENT.PIN}`,
+      debounce({
+        callback: () => toggleSidebar(false),
+        // Mousemove handles the edge case in which mouseleave and then immediately mouseenter
```

Comment:
When user mouses out, it will trigger the timer. then user hovers back to the sidebar. We need to cancel the timer when user hovers back. debounce might be a misleading name. The purpose of listening the `mousemove` ('cancel' event) is cancelling timer if there is any. 

Beside "invoke after delay", the function also resets the timer in demand. I name the function `debounce` because it basically invokes the last `non-cancel` event. I realized the name is misleading. I will try to think another approach.

Reply:
@nphi1212 any thought on my suggestion above?

Reply:
@buunguyen  I agreed with you what it would be better to keep the logic explicitly in one place.  Will try to implement it =) Many thanks for your valued feedbacks